consensus/src/quorum_store/utils.rs

```diff
@@ -295,13 +302,24 @@ impl ProofQueue {
                     .expect("Entry for unexpired digest must exist")
                 {
                     Some(proof) => {
-                        cur_bytes += proof.info().num_bytes;
-                        cur_txns += proof.info().num_txns;
-                        if cur_bytes > max_bytes || cur_txns > max_txns {
-                            // Exceeded the limit for requested bytes or number of transactions.
-                            break;
+                        match batch_reader.get_batch_optimistic(proof.clone()).await {
+                            Some(txns) => {
+                                let mut unexpired = 0;
+                                txns.iter().for_each(|txn| {
+                                    if txn.expiration_timestamp_secs() < curr_time.as_secs() {
+                                        unexpired += 1;
+                                    }
+                                });
+                                cur_bytes += proof.info().num_bytes;
```

Comment:
Are you counting bytes for all transactions? not only unexpired?

Reply:
For the experiments, we don't hit this limit. In practice, I think we should count all txns otherwise block might be too big.

Reply:
Ok, so please make a comment to explain this.